src/axelrod_dojo/archetypes/cycler.py

```diff
@@ -0,0 +1,142 @@
+    """
+    Cycler params is a class to aid with the processes of calculating the best sequence of moves for any given set of
+    opponents. Each of the population in our algoruthem will be an instance of this class for putting into our
+    genetic lifecycle.
+
+    **********
+    TODOs:
+        * create a metaheuristic for mutations to increase mutation & crossover efficiency (ie, dont change the same
+        gene 2 times in a row or try and do a crossover if the genes being swapped are sufficiently different)
+    """
+
+    def __init__(self, sequence=None, sequence_length: int = 200, mutation_probability=0.1, mutation_potency=1):
+
+        if sequence is None:
+            # generates sequence uses a map to the Action class to change a sequence of 1 & 0s to C & D actions
+            self.sequence = self.generate_random_sequence(sequence_length)
+            self.sequence_length = sequence_length
+        else:
+            #  when passing a sequence, make a copy of the sequence to ensure mutation is for the instance only.
+            self.sequence = list(sequence)
+            self.sequence_length = len(sequence)
+
+        self.mutation_probability = mutation_probability
+        self.mutation_potency = mutation_potency
+
+    def __repr__(self):
+        return "{}".format(self.sequence)
+
+    @staticmethod
+    def generate_random_sequence(sequence_length):
+        """
+        Generates a sequence of random moves when an instance is initialised
+
+        Parameters
+        ----------
+        sequence_length - length or random moves to generate
+
+        Returns
+        -------
+        list - a list of C & D actions: e.g. [C,C,D,D,C]
+        """
+        return list(map(axl.Action, random.randint(0, 1 + 1, (sequence_length, 1))))
+
+    def crossover(self, other_cycler):
+        """
+        creates and returns a new CyclerParams instance with a single crossover point in the middle
+
+        Parameters
+        ----------
+        other_cycler - the other cycler where we get the other half of the sequence
+
+        Returns
+        -------
+        CyclerParams
+
+        """
+        seq1 = self.get_sequence()
+        seq2 = other_cycler.get_sequence()
+
+        midpoint = int(len(seq1) / 2)
+
+        # If the length is odd, keep more from sequence 1
+        if len(seq1) % 2 == 1:
+            midpoint += 1
+
+        new_seq = seq1[:midpoint] + seq2[midpoint:]
+
+        return CyclerParams(sequence=new_seq)
+
+    def mutate(self):
+        """
+        Basic mutation which may change any random gene(s) in the sequence.
+        """
+        # if the mutation occurs
+        if random.rand() <= self.mutation_probability:
+            mutated_sequence = self.get_sequence()
+            for _ in range(self.mutation_potency):
+                index_to_change = random.randint(0, len(mutated_sequence))
+                # Mutation - change a single gene
+                if mutated_sequence[index_to_change] == C:
```

Comment:
I think we have a flip_dict from Axelrod that maps C -> D and D -> C

Reply:
They're methods on the instances:

```python
>>> import axelrod as axl
>>> C = axl.Action.C
>>> C.flip()
D
```

Reply:
Even better :)